### ISSUE TITLE
mast: photometer

### ISSUE THREAD

**Issue by scruzin** [ausocean/rig#4]
At present, we have no direct way of measuring solar radiance (light intensity).

While the PV voltage loosely correlates with solar radiance, it cannot be relied on to measure the latter. Further more, there are times when panels are occluded and/or damaged, even when solar radiance is high. Having an independent way to measure solar radiance would enable us to detect such situations.

It is therefore proposed to mount a _photometer_ inside the dome on top of the nav lights,  at the very top of mast.

**Comment by Breezethebluemaskman** [comment]
could be a silly question, but would a photometer pick up the light from the LEDs and potentially distort the data? or is there a particular wavelength that photometers are concerned with that will have nil impact from the LEDs?

**Comment by scruzin** [comment]
I don't think so. The photometer would be aimed up, above the nav light LEDs. However, if there's stray light, we can mount the photometer inside a small upward-facing cone.

Also, the nav light LEDs should only come on at night (although we don't always bother).

**Comment by ao-david** [comment]
How precise are we looking to be? From my brief research it seems like to get a good indication of solar irradiance we need a bit more circuitry then just a single sensor ([See here](https://www.re-innovation.co.uk/docs/low-cost-diy-solar-irradiance-sensor-for-pv-system-monitoring/)). However, if we just want to measure brightness (across different wavelengths) then we could consider an option like [this](https://www.adafruit.com/product/1980).

We need to decide on this component before I can finalise the new PCB design. If we are happy that the sensor will just be analog, I can leave an analog pin free for future development. However if we need another interface, we need to consider this before I can finish the design.

**Comment by scruzin** [comment]
I think that Adafruit TSL2591 fits the bill nicely.  As an I2C device, we could potentially integrate it with the existing ESP8266, not just the ESP32.

Also, at 19mm x 16mm, it looks like it is small enough to sit on top of the mast light LEDs. We'll just need to design a nice holder for it that integrates with the latter.

**Comment by ao-david** [comment]
> I think that Adafruit TSL2591 fits the bill nicely. As an I2C device, we could potentially integrate it with the existing ESP8266, not just the ESP32.
> 
> Also, at 19mm x 16mm, it looks like it is small enough to sit on top of the mast light LEDs. We'll just need to design a nice holder for it that integrates with the latter.

I will go ahead and order a couple for testing

**Comment by ao-david** [comment]
I have received the light sensors for testing, and got them running with the ESP32 dev board that I have on hand. 
## Wiring
SDA - GPIO21 (ESP32 I2C default)
SCL - GPIO22 (ESP32 I2C default)
## Firmware
I have tested with the example program supplied by adafruit [here](https://learn.adafruit.com/adafruit-tsl2591/wiring-and-test). Which gives the following serial output from my desk:
```
[ 665001 ms ] IR: 5503  Full: 13122  Visible: 7619  Lux: 240.654953
[ 665863 ms ] IR: 5267  Full: 12641  Visible: 7374  Lux: 234.004257
[ 666725 ms ] IR: 4009  Full: 9834  Visible: 5825  Lux: 187.698395
[ 667587 ms ] IR: 3840  Full: 9391  Visible: 5551  Lux: 178.496414
[ 668449 ms ] IR: 3544  Full: 8749  Visible: 5205  Lux: 168.454254
[ 669311 ms ] IR: 2974  Full: 7434  Visible: 4460  Lux: 145.561340
[ 670173 ms ] IR: 2709  Full: 6812  Visible: 4103  Lux: 134.439636
[ 671035 ms ] IR: 2638  Full: 6647  Visible: 4009  Lux: 131.536224
[ 671897 ms ] IR: 2611  Full: 6583  Visible: 3972  Lux: 130.374771
[ 672759 ms ] IR: 2651  Full: 6678  Visible: 4027  Lux: 132.103928
[ 673621 ms ] IR: 2767  Full: 7069  Visible: 4302  Lux: 142.423523
[ 674483 ms ] IR: 2877  Full: 7221  Visible: 4344  Lux: 142.161224
[ 675345 ms ] IR: 2826  Full: 7108  Visible: 4282  Lux: 140.328156
[ 676207 ms ] IR: 2832  Full: 7122  Visible: 4290  Lux: 140.576111
[ 677069 ms ] IR: 2733  Full: 6887  Visible: 4154  Lux: 136.301865
```
